src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -134,11 +128,19 @@ public Object invokeMethod(String name, Object args) {
             if (exec==null)
                 exec = (CpsFlowExecution) handle.get();
         } catch (IOException e) {
-            throw new Error(e); // TODO
+            throw new GroovyRuntimeException(e);
         }
 
         if (functions == null) {
             functions = new TreeMap<>();
+            while (StepDescriptor.all().isEmpty()) {
```

Comment:
Could, but it should not really matter since just obtaining the `ExtensionList` instance is cheap—and anyway we are only doing this once per build.

(Really I considered getting rid of the cache altogether, since the list is not very long and `getFunctionName` is generally instantaneous. Was just not sure how much overhead there is in iterating the list.)